### PR TITLE
New methods for fetching all user settings and all feature flags

### PR DESCRIPTION
**Pull request by jkodumal**
If streaming is enabled, these new methods will not make remote calls

### CODE REVIEW

**Comment by jkodumal** [comment]
HoundCI seems to be having issues unrelated to these changes.

**Comment by pkaeding** [comment]
:+1:

File: lib/ldclient-rb/ldclient.rb [hunk 1]
```diff
@@ -203,25 +203,38 @@ def track(event_name, user, data)
...
+          JSON.parse(res.body, symbolize_names: true)         
+        else
+          @config.logger.error("[LDClient] Unexpected status code #{res.status}")
+          Hash.new
+        end
       end
     end
 
+    def get_user_settings(user)
+      Hash[all_flags.map { |key, feature| [key, evaluate(feature, user)]}]
```
Comment (pkaeding) [inline lib/ldclient-rb/ldclient.rb]: I think we should send feature events here.

**Comment by slumos** [comment]
Almost have this integrated, but wanted to mention that I ran into the same problem mentioned at https://github.com/plataformatec/faraday-http-cache/issues/38. There's a gem that supposedly works around the problem that ultimately lies with Net::HTTP, [https://github.com/ma2gedev/faraday-encoding](faraday-encoding), but for now I just changed the serializer to Marshal as in [https://github.com/andrew/contribulator/pull/47/files](this other project).

I also added back our `LDClient#add_user_override` to my fork. Not sure if you were planning to implement that, but if not then I'll move it elsewhere so we can use your gem unmodified.

**Comment by jkodumal** [comment]
@slumos Happy to take both changes (the add_user_override and the Marshal serializer) if you submit a PR. Thanks!

**Comment by slumos** [comment]
This is working well for us.

**Comment by slumos** [comment]
Ping @jkodumal. Will this get merged soon?

**Pull request merged**